core/common/src/main/java/alluxio/util/io/PathUtils.java

```diff
@@ -283,17 +290,27 @@ public static String subtractPaths(String path, String prefix) throws InvalidPat
    * @throws InvalidPathException when the path or prefix is invalid
    */
   public static boolean hasPrefix(String path, String prefix) throws InvalidPathException {
-    String[] pathComponents = getPathComponents(path);
-    String[] prefixComponents = getPathComponents(prefix);
-    if (pathComponents.length < prefixComponents.length) {
-      return false;
+    if (prefix.isEmpty() || path.isEmpty()) {
+      throw new InvalidPathException(ExceptionMessage.PATH_INVALID.getMessage(""));
     }
-    for (int i = 0; i < prefixComponents.length; i++) {
-      if (!pathComponents[i].equals(prefixComponents[i])) {
-        return false;
+
+    if (prefix.equals("/")) {
+      return true;
+    }
+
+    if (prefix.charAt(prefix.length() - 1) == '/' && path.charAt(path.length() - 1) != '/') {
+      path = path + '/';
+      if (path.equals(prefix)) {
+        return true;
       }
     }
-    return true;
+
+    if (path.length() < prefix.length()) {
+      return false;
+    }
+    return path.startsWith(prefix)
+        &&
+        (path.length() == prefix.length() || path.charAt(prefix.length()) == '/');
```

Comment:
Can you please add a few comments here to explain this complex logic expression? Or, you can break it into several early return if conditions, as above.